python/cuml/tests/test_pickle.py

```diff
@@ -554,6 +554,35 @@ def assert_model(pickled_model, X_train):
     pickle_save_load(tmpdir, create_mod, assert_model)
 
 
+@pytest.mark.parametrize('datatype', [np.float32, np.float64])
+@pytest.mark.parametrize('keys', hdbscan_model.keys())
+@pytest.mark.parametrize('data_size', [unit_param([500, 20, 10]),
+                                       stress_param([500000, 1000, 500])])
+def test_hdbscan_pickle(tmpdir, datatype, keys, data_size):
+    result = {}
+    from cuml.cluster.hdbscan.prediction import all_points_membership_vectors
+    from cuml.cluster.hdbscan.prediction import approximate_predict
+
+    def create_mod():
+        nrows, ncols, n_info = data_size
+        X_train, _, _ = make_dataset(datatype, nrows, ncols, n_info)
+        model = hdbscan_model[keys](prediction_data=True)
```

Comment:
Just to make sure we've covered all cases, we should probably generate w/ `prediction_data=True|False` and also make sure we invoke the plotting API / force build the condensed hierarchy (essentially make sure the state is both tested populated and unpopulated for any pointers on the hdbscan cython object) to verify that these won't cause a segmentation fault.